content/reference/gql-types.md

```diff
@@ -0,0 +1,122 @@
+---
+title: GraphQL types and filters
+
+description: >-
+  A reference of the data types in the Rhize API and of the filters available for each type.
+categories: ["reference"]
+weight: 930
+menu:
+  main:
+    name: GQL types and filters
+    identifier: gql-filters
+    parent: reference
+---
+
+This page provides a reference of the data types enforced by the Rhize database schema,
+and of the filters that can apply to these types when you query, update, or delete a set of resources.
+For an extended guide, with examples, read [Use query filters]({{< relref "/how-to/gql/filter" >}}).
+
+{{< notice "note" >}}
+These filters are based on Rhize's implementation of the Dgraph [`@search` directives](https://dgraph.io/docs/graphql/schema/directives/search/).
+{{< /notice >}}
+
+## Data types
+
+Every object in the Rhize schema has fields that are of one of the basic data types.
+From the other perspective, these data types define fields that compose manufacturing objects,
+objects defined precisely by ISA-95 and enforced by Rhize's database schema.
+
+### Basic types
+
+Every manufacturing object in the Rhize database is made of fields that are of one the following basic types.
+In official GraphQL terminology, these types are called [_scalar types_](https://graphql.org/learn/schema/#scalar-types).
+
+- `String`
+- `Int`
+- `Float`
+- `Enum`
+- `Boolean`
+- `id`: A string representing a unique object within a defined [object type](#object-type).
+- `iid`: The object's unique address in the database. For example, `0xf9b49`.
+- `DateTime`: A timestamp in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format.
+- `Geo`: Geometry types for geo-spatial coordinates
+
+### Object type
+
+The preceding basic types form the building blocks for Rhize's manufacturing object schema, with data models corresponding to ISA-95.
+
+Each object is made of manufacturing-specific fields of one of the basic types.
+For example, the `materialActual`  object has basic fields, including:
+- `description`, a `String`.
+- `effectiveEnd`, a `DateTime`
+
+The `materialActual` also has complex fields describing associated manufacturing objects.
+For example, its fields include
+the array of associated `MaterialLot` objects, the `MaterialDefinition` object, and so on.
+All objects in the database have relationships to other objects.
+
+## Scalar filters
+
+Every field that has a basic type can be used as a filter.
+The filters that are available depend on the data type, but the behavior of the `String` filters corresponds closely to `DateTime` and `Int` filters.
```

Comment:
This doesn't seem true based on a quick attempt. The filter on `queryWorkflowSpecification` allows me to filter an `id` by the methods listed under "String filters", but not for example the `_modifiedBy` field. 

My understanding is that these filters are defined as `@search(by: ["foo"])` directives in the schema (and potentially also by `@id`).

To see this effect in action, compare the filter for `workflowSpecification` to that of `Rule`. The former does not have `@search` on `_modifiedBy` and thus can't be filtered by that field. The latter does.

![filters](https://github.com/libremfg/libremfg.github.io/assets/122067969/aaf9f7d8-dbe0-4acf-9380-74bb91080952)

Reply:
Whoa! Good catch. That would have been a _huge_ inaccuracy. I think I just had the "bad luck" to do all my testing on objects with lots of filters.